### DISCUSSION TITLE
Enabling of deprecated cipher suites in OpenSSL 3.2.1 ("TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256")

### DISCUSSION THREAD

**Discussion by VeronikaNguyen** [openssl/openssl#25514] [Q&A] [answered]
We are attempting to enable the deprecated cipher suite `TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256` in OpenSSL 3.2.1 and use it in TLS 1.2. The compile option `enable_weak_ssl_ciphers` is set to `True`. However, when calling `SSL_CTX_set_cipher_list(sslCtx, "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256")`, we encounter the following error:

`6C980000:error:0A0000B9:SSL routines:SSL_CTX_set_cipher_list:no cipher match:ssl\ssl_lib.c:3358:`

Does OpenSSL 3.2.1 still support the `TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256` cipher suite? What compiler flags are required to enable this cipher suite?

**Comment by kroeckx** [comment]
I think we no longer support static ECDH at all. Is there a reason you want to use static ECDH, instead of ECDHE?

We have never seen a non-test certificate that has static ECDH parameters in it.

Maybe you're looking for TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 (ECDHE-ECDSA-AES128-SHA256) instead?

**Reply by VeronikaNguyen** [reply]
Thanks for the answer!

We are working on a test tool and previously used mbedTLS which still supported this cipher suite. Some users of this tool still need this deprecated cipher suite.

**Reply by kroeckx** [reply, marked as answer]
So I'm wondering what the use of case of using static (EC)DH is. In case of the ephemeral (EC)DHE, both sides generate a public/private key pair, that changes for each connection. In case of (EC)DH, the server has a fixed public/private key, of which the public part is embedded in the certificate. When using client authenticate, the client could also have a certificate with the static parameters in it.

Support for it seems to have been removed in 2016 in commit ce0c1f2bb2fd296f10a2847844205df0ed95fb8e.

So I'm wondering why you use it. Is this some performance thing?

**Reply by VeronikaNguyen** [reply]
Out of compatibility reasons. Some old systems which are tested still use it which is why we wanted to support this cipher suite.

**Comment by TWLemontree** [comment]
Did you ultimately resolve this issue? How did you resolve it?
